src/unix/stream.c

```diff
@@ -837,6 +841,30 @@ static void uv__write(uv_stream_t* stream) {
 #else
     while (n == -1 && errno == EINTR);
 #endif
+#ifdef MSG_FASTOPEN
+  } else if (stream->flags & UV_TCP_FASTOPEN) {
+    struct sockaddr *addr = { 0 };
+    socklen_t addrlen;
+    if (getsockname(uv__stream_fd(stream), addr, &addrlen)) {
+      n = -1;
+    } else {
+      do {
+        n = sendto(uv__stream_fd(stream), iov[0].iov_base, iov[0].iov_len,
+                   MSG_FASTOPEN, addr, addrlen);
+      }
+#  if defined(__APPLE__)
+      /*
+       * Due to a possible kernel bug at least in OS X 10.10 "Yosemite",
+       * EPROTOTYPE can be returned while trying to write to a socket that is
```

Comment:
This is all over the code around, originally reported here: https://github.com/libuv/libuv/pull/482